### ISSUE TITLE
failed to list all snapshots

### ISSUE THREAD

**Issue by Madhu-1** [gluster/glusterd2#839]
listing snapshots failed with below messages

```
curl http://dhcp42-235:24007/v1/snapshots 
{"errors":[
               {"code":1,"message":"EOF"}
       ]
}
```

as volume name is not mandatory for listing the snapshots, This request should get passed.

@aravindavk @prashanthpai want to know your thoughts on this.

**Comment by rafikc30** [comment]
@Madhu-1 

Snapshot list command accepts a req variable, since req is not given here, the parsing is failed. Fixed the same as part of the commit https://github.com/gluster/glusterd2/pull/846/commits/984c98b138a0bc619bdf536e94dd2d563ec5ef54

**Comment by Madhu-1** [comment]
@rafikc30, what happens if I pass invalid request? as per your commit, the validation passes (the validation should fail for an invalid request), we should avoid using Request body for HTTP GET requests.

**Comment by rafikc30** [comment]
I understand. To list all snapshot and to list snapshots of a single volume, What do you suggest to use? I was using two rest endpoints like /v1/snapshots and /v1/snapshots/volume/{volumename} . But this was also not good, and kaushal suggested to use HTTP GET requests.

So any other suggestions?

**Comment by Madhu-1** [comment]
@rafikc30 I would suggest using query param instead of request body for this kind of request. @kshlm  @prashanthpai @aravindavk we need some more info on this.

**Comment by Madhu-1** [comment]
fixed in #846
